### ISSUE TITLE
Travis issue

### ISSUE THREAD

**Issue by BenFradet** [snowplow/nsq-spark-example-project#1]
Can we turn on travis on for this repo @alexanderdean ?

Also can you detail the issue @aldemirenes ?

**Comment by alexanderdean** [comment]
Travis has been enabled now, apologies for the delay

**Comment by aldemirenes** [comment]
@BenFradet Spark job does not start in travis build. You can see the last output of the Spark job in [here](https://travis-ci.org/snowplow-incubator/nsq-spark-example-project/builds/287575241#L1905).
I think this is related to the number of cores which used by Spark job. While using the Spark streaming, at least two CPU cores must be allocated to the Spark job. You can see the related discussion in this [Stackoverflow thread](https://stackoverflow.com/a/28052228/5265314). However, I think that machine instance which Spark job is running have at least 3 cores in order to work successfully because 2 cores will be allocated to the Spark job already. There should be at least one more core to continue the other works. But, travis instances have at most 2 CPU cores as [specified in this page](https://docs.travis-ci.com/user/reference/overview/). I can not find any solution to this problem yet. Any suggestion is welcomed.

**Comment by BenFradet** [comment]
@aldemirenes I don't think this is the issue, I've run tests on travis with only 1 allocated core to spark while setting up zk and kafka programmatically without issues. Can't you run nsq as part of your spec?

**Comment by aldemirenes** [comment]
@BenFradet I reached this conclusion about the number of cores after running the Spark job in the Vagrant env with 2 and 3 cores. While Spark job is running completely fine in the env with 3 cores, it is stopping in the same part with travis build when Spark job is running in the env with 2 cores. If it is okay, could you check the source code if everything seems to okay to you? Project does not have so much code, checking it should be easy. By the way, can you explain what you mean with 'running nsq as part of spec', please?

**Comment by BenFradet** [comment]
I meant launching it in your program not as a script, for example [this](https://github.com/BenFradet/spark-kafka-writer/blob/master/src/test/scala/com/github/benfradet/spark/kafka/writer/KafkaTestUtils.scala) launches a zookeeper and a kafka during the tests not outside, this makes the tests reliable and self-contained.

**Comment by aldemirenes** [comment]
I was looking for it before but I could not find any library for starting nsq server programmatically. I can look for it one more time if you think that problem is related with this.

**Comment by BenFradet** [comment]
Since you're running only one receiver, I guess it should be fine with 2 cores, I don't get why we'd need 3.

Could you try with `local[*]` and let Spark figure it out itself in https://github.com/snowplow-incubator/nsq-spark-example-project/blob/master/src/test/scala/com/snowplowanalytics/spark/WordCountTest.scala#L70 ?

**Comment by BenFradet** [comment]
Also could you try with `sudo: required` since according to [this](https://docs.travis-ci.com/user/reference/overview/) we might get more than 2 cores.

**Comment by aldemirenes** [comment]
I tried `local[*]` and `sudo: required`. There is no success with the neither of the two.

**Comment by BenFradet** [comment]
Then I guess we are stuck

**Comment by alexanderdean** [comment]
Huh, shame
